src/server/middleware/v2auth/auth.go

```diff
@@ -83,25 +78,31 @@ func (rc *reqChecker) projectID(name string) (int64, error) {
 	return p.ProjectID, nil
 }
 
-func getAction(req *http.Request) rbac.Action {
-	pushActions := map[string]struct{}{
-		http.MethodPost:   {},
-		http.MethodDelete: {},
-		http.MethodPatch:  {},
-		http.MethodPut:    {},
+func getChallenge(req *http.Request, accessList []access) string {
```

Comment:
`Basic` and  `Bearer` should returned together in the challenge so that cli will first try to use `Basic` auth, then `Bearer` auth.

I think most time the `Basic` auth will be ok, and the anonymous access will use the `Bearer` auth. (This needs to be confirmed)

Reply:
Return basic auth will fail #11266